transform-message-formats/README.md

```diff
@@ -0,0 +1,457 @@
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    isolated resource function post categories/[string category]/reserve(HealthcareReservation payload)
+            returns ReservationResponse|http:NotFound|http:BadRequest|http:InternalServerError {
+        
+   }
+}
+```
+
+- Use `/healthcare` as the service path (or the context) of the service which is attached to the listener listening on port `port`. 
+
+- The HTTP resource allows the `POST` operation on resource path `/categories/{category}/reserve`, where `category` (corresponding to the specialization) is a path parameter.
+
+- Use `HealthcareReservation` as a parameter indicating that the resource expects a JSON object corresponding to `HealthcareReservation` as the payload. 
+
+- Use `ReservationResponse|http:NotFound|http:InternalServerError` as the return type to indicate that the response will have a JSON payload corresponding to `ReservationResponse` on success or the response will be a "NotFound", or "InternalServerError" response on error.
+
+8. Implement the logic.
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    isolated resource function post categories/[string category]/reserve(HealthcareReservation payload)
+            returns ReservationResponse|http:NotFound|http:InternalServerError {
+        HospitalReservation hospitalReservation = transform(payload);
+
+        ReservationResponse|http:ClientError resp =
+                    hospitalServiceEP->/[payload.hospital_id]/categories/[category]/reserve.post(hospitalReservation);
+
+        if resp is ReservationResponse {
+            log:printDebug("Reservation request successful",
+                            name = payload.name,
+                            appointmentNumber = resp.appointmentNumber);
+            return resp;
+        }
+
+        log:printError("Reservation request failed", resp);
+        if resp is http:ClientRequestError {
+            return <http:NotFound> {body: "Unknown hospital, doctor or category"};
+        }
+
+        return <http:InternalServerError> {body: resp.message()};
+    }
+}
+```
+
+- Use the `transform` function to transform the payload from a `HealthcareReservation` record to a `HospitalReservation` record.
+
+    ```ballerina
+    HospitalReservation hospitalReservation = transform(payload);
+    ```
+
+- Use the transformed payload in the request to the hospital service and get the response. Here, The `hospital_id` and `category` values are used as path parameters.
+
+    ```ballerina
+    ReservationResponse|http:ClientError resp =
+                    hospitalServiceEP->/[payload.hospital_id]/categories/[category]/reserve.post(hospitalReservation);
+    ```
+
+- The `log` functions are used to [log](https://ballerina.io/learn/by-example/#log) information at `INFO`, `DEBUG`, and `ERROR` log levels.
+
+- Use the `is` check to check whether the response is `ReservationResponse` and return it as is (i.e., reservation successful).
```

Comment:
```suggestion
- Use the `is` check to check whether the response is a `ReservationResponse` and return it as is (i.e., reservation successful).
```

Reply:
If we put "a" we have to say "a `ReservationResponse` record"

Reply:
Yeah. +1 without `a`

Reply:
fixed via https://github.com/ballerina-guides/integration-tutorials/pull/8/commits/4c858f776cb3ce0a643ec14ad4c0d69b3afa903c